### PR TITLE
Add support for using gcs instead of s3

### PR DESCRIPTION
**Pull request by relud**
co-depends on https://github.com/mozilla/telemetry-airflow/pull/1515

### CODE REVIEW

File: probe_scraper/remote_storage/gcs.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from .utils import compile_glob, extract_bucket_and_prefix, path_walk
+
+
+def pull(src: str, dst: Path):
+    bucket_name, prefix = extract_bucket_and_prefix(src)
+    prefix_length = len(prefix)
+    bucket = storage.Client().get_bucket(bucket_name)
+    # TODO verify that this correctly handles prefix being a file
+    for blob in bucket.list_blobs(prefix=prefix):
```
Comment (akkomar) [inline probe_scraper/remote_storage/gcs.py:14]: Would it make sense to use `gsutil rsync` to simplify this, similar to `aws s3 sync` in the other backend?

File: probe_scraper/remote_storage/__init__.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def _get_implementation(remote: str):
+    if remote.startswith("s3://"):
+        return s3
+    elif remote.startswith("gs://"):
+        return gcs
+    else:
+        raise ValueError(
+            f"remote path must have scheme like s3:// or gs://, got: {remote!r}"
```
[the remark this answers is not in the document]
Comment (akkomar) [reply probe_scraper/remote_storage/__init__.py:20]: Also this requires a change in a DAG in `telemetry-airflow`.

**Review by akkomar — COMMENTED** [review]
What's the plan for this switch?
What about schema generator which is consuming artifacts produced here?
What about Cloudfront?

File: probe_scraper/remote_storage/gcs.py (hunk 1 above)
Comment (relud) [reply probe_scraper/remote_storage/gcs.py:14]: Unfortunately `gsutil` doesn't expose flags for setting content-encoding and content-type, but the sdk does.

**Comment by relud** [comment]
> What's the plan for this switch? What about schema generator which is consuming artifacts produced here? What about Cloudfront?

We will continue to use s3 in production while push-mode is being tested. When push mode is ready for production, that's when we'll cut the probe info url over to gcs. schema generator will need to be updated to use gcs at that time if it's not using the probe info url. Cloudfront will presumably be replaced with something in GCP, but I expect that will all be handled by SRE.

File: probe_scraper/remote_storage/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (relud) [reply probe_scraper/remote_storage/__init__.py:20]: https://github.com/mozilla/telemetry-airflow/pull/1515

File: probe_scraper/remote_storage/gcs.py (hunk 1 above)
Comment (relud) [reply probe_scraper/remote_storage/gcs.py:13]: turns out it does support those things, via `-h`. I also thought that gsutil wasn't a python package, but it turns out I was also wrong about that.

I'm working on changing this to use `gsutil`

**Review by akkomar — APPROVED** [review]

**Pull request merged**